android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/webextension/WebExtensionRuntime.kt

```diff
@@ -37,6 +37,29 @@ interface WebExtensionRuntime {
         return CancellableOperation.Noop()
     }
 
+    /**
+     * Installs an [WebExtension] from the provided [url] in this engine.
+     *
+     * @param url the url pointing to either a resources path for locating the extension
+     * within the APK file (e.g. resource://android/assets/extensions/my_web_ext) or to a
+     * local (e.g. resource://android/assets/extensions/my_web_ext.xpi) or remote
+     * (e.g. https://addons.mozilla.org/firefox/downloads/file/123/some_web_ext.xpi) XPI file.
+     * @param onSuccess (optional) callback invoked if the extension was installed successfully,
+     * providing access to the [WebExtension] object for bi-directional messaging.
+     * @param onError (optional) callback invoked if there was an error installing the extension.
+     * This callback is invoked with an [UnsupportedOperationException] in case the engine doesn't
+     * have web extension support.
+     */
+    @Suppress("LongParameterList")
+    fun installWebExtension(
```

Comment:
I think it will be a significant one, with have around 37 usages of this signature, and all their dependencies that will need changes, without counting which other issue we could find. I would rather prefer to file a follow up bug, land this functionality and start collecting feedback, while we explore the possible refactor. 

<img width="874" alt="image" src="https://github.com/mozilla-mobile/firefox-android/assets/773158/a074c53c-bf48-43fb-bca3-c159da786c7d">

Reply:
aren't these occurrences mostly in test files?

Reply:
We have around 10 of them that aren't tests files but those occurrences have tests and dependencies, changing them will require updating their dependencies. As the change is a refactor that it's not related to the functionality, I think we could land the functionality start collecting feedback and discovering possible bugs, while simultaneously addressing the possible refactor and separated ticket.

Reply:
I am OK with deferring the refactoring to a follow-up bug but I don't think we should introduce a new method that we will remove shortly after. Maybe we should refactor the code first then, and rebase this PR on top of the refactored codebase?

Reply:
The main issue that I see is that we won't know the extend of the refactor until we do it, as it's a **possible refactor**. Meanwhile, we're blocking a feature which is ready to land to give a try to a **possible refactor**,  which block us from start collecting feedback early.  Normally in Fenix we prioritize an agile process, where we try to have small incremental patches which speed up the feedback loop between in detecting issues.

Also keeping a PR opened have a maintenance cost, which is the burner of maintain it from being stale, as the source code is alive and keeps changing.

An example of keeping the PR from being stale and it's this merge conflict. 
<img width="867" alt="image" src="https://github.com/mozilla-mobile/firefox-android/assets/773158/4bc63c05-12bd-48ce-9a6d-a1e982ab63ab">

Reply:
Ok, so I started the refactoring and it looks like the way built-ins are treated is going to annoy us... https://searchfox.org/mozilla-central/rev/9bc264fbc5d6e618d8f3b9677a8f5e8550b94dbc/mobile/android/modules/geckoview/GeckoViewWebExtension.sys.mjs#903,911

I guess that forces me to reconsider this patch 😛

Reply:
How should we proceed?  😛

Reply:
Let's file a follow-up bug to refactor the `install` methods: I don't think the `id` is used for normal extensions, only for built-in ones.